src/Group.php

```diff
@@ -18,11 +22,38 @@ class Group
      * @var array<int,User>
      */
     private array $members;
+    private Configuration $graphApiConfig;
+    private Client $guzzle;
+    private string $serviceUrl;
+    private string $accessToken;
+
+    /**
+     * @phpstan-param array{
+     *                      'headers'?:array<string, mixed>,
+     *                      'verify'?:bool,
+     *                      'webfinger'?:bool,
+     *                      'guzzle'?:\GuzzleHttp\Client
+     *                      } $connectionConfig
+     */
+    private array $connectionConfig; /** @phpstan-ignore-line */
+
     /**
      * @param OpenApiGroup $openApiGroup
+     * @param string $serviceUrl
+     * @phpstan-param array{
+     *                      'headers'?:array<string, mixed>,
+     *                      'verify'?:bool,
+     *                      'webfinger'?:bool,
+     *                      'guzzle'?:\GuzzleHttp\Client
+     *                      } $connectionConfig
+     * @param string $accessToken
      */
-    public function __construct(OpenApiGroup $openApiGroup)
-    {
+    public function __construct(
+        OpenApiGroup $openApiGroup,
+        string $serviceUrl = "",
+        array $connectionConfig = [],
+        string &$accessToken = ""
+    ) {
```

Comment:
I think these parameters are not optional (except of `connectionConfig`)
```suggestion
    public function __construct(
        OpenApiGroup $openApiGroup,
        string $serviceUrl,
        array $connectionConfig,
        string &$accessToken
    ) {
```